docs/architecture/Data_caching_lineage.md

```diff
@@ -0,0 +1,297 @@
+1. *When possible*, label files are placed according to their primary purpose within the training process (e.g. `train`, `test`).
+
+> If a random runtime shuffling/splitting is used, for a purpose of experiment reproducibility the split information (file list for every group) must be stored in the run's output storage, and reproduced in derived runs.
+
+Here are a few examples with dataset name `object_detection` and labels in `.csv` format.
+
+**Example 1** - annotations are stored by the data category
+
+```text
+[object_detection]
+└── [train]
+    ├── {location_name}.{camera_id}.029347623462.csv
+    ├── {location_name}.{camera_id}.029347346341.csv
+    └── {location_name}.{camera_id}.235235994566.csv
+└── [test]
+    ├── {location_name}.{camera_id}.029347623410.csv
+    ├── {location_name}.{camera_id}.029347346395.csv
+    └── {location_name}.{camera_id}.235232302300.csv    
+└── [validation]
+    ├── {location_name}.{camera_id}.029347678464.csv
+    ├── {location_name}.{camera_id}.462747346320.csv
+    └── {location_name}.{camera_id}.556435994562.csv      
+```
+
+**Example 2** - annotations are stored by the model stage category
+
+```text
+[object_detection]
+└── [train] <- for runtime train/test/val split
+    ├── {location_name}.{camera_id}.029347623462.csv
+    ├── {location_name}.{camera_id}.029347346341.csv
+    └── {location_name}.{camera_id}.235235994566.csv
+└── [qa_validation] <- for QA validation tests
+    ├── {location_name}.{camera_id}.029347623410.csv
+    ├── {location_name}.{camera_id}.029347346395.csv
+    └── {location_name}.{camera_id}.235232302300.csv      
+```
+
+**Example 3** - annotations are stored by the model stage category and time
+
+> **This is the recommended label storage structure**
+
+```text
+[object_detection]
+└── [train] <- for runtime train/test/val split
+│   └── [{year}.{month}]
+│   │   ├── {location_name}.{camera_id}.029347623462.csv
+│   │   ├── {location_name}.{camera_id}.029347346341.csv
+│   │   └── {location_name}.{camera_id}.235235994566.csv
+└── [qa_validation] <- for QA validation tests
+    ├── {location_name}.{camera_id}.029347623410.csv
+    ├── {location_name}.{camera_id}.029347346395.csv
+    └── {location_name}.{camera_id}.235232302300.csv      
+```
+
+There are labels that are yet to be verified, such as those made using transfer learning or another auto-labeling technique. For these labels, the storage structure needs to reflect the state in the data collection and verification processes. On the data collection side, the source (location, camera id) should be presented in the structure. On the verification side, the state of the labels should be presented: auto-labeled as 'positive', auto-labeled as 'negative', and 'verified'. All verified labels must be further processed/converted and transferred to the `train` (default) or `qa_validation` subsets.
```

Comment:
What does "positive" mean here? Is this a binary classification problem, or are you using it in some other semantic context?

Reply:
This piece is hard to get without context. We had an auto-labeling process with `positive` and `negative` images as auto-labeled images with applicable objects or without them respectfully. I suggest to remove this piece entirely (`Data (Images) and Annotations directory structure`).

Reply:
I really like laying out a strategy to deal with untrusted labels, so would like to have it in there, perhaps just a brief problem framing at the beginning (e.g. "For these labels" => "For these labels (for this example, we'll use 'positive/negative')")

Reply:
Reworded it a bit to call out the object detect example or image classification. How does it look?

Reply:
I'm not seeing the rewording, did it get committed/pushed?

Reply:
The highlighted part is what I added:  "...On the verification side, the state of the labels should be presented: **_for example, if we are doing object detection for trucks or binary classification,_** the images can be auto-labeled as 'positive', auto-labeled as 'negative', and 'verified'..."